### PR TITLE
Include annotations in Arguments as defined in the Thrift file

### PR DESCRIPTION
**Pull request by dmtsui**
Currently annotations defined in the Thrift file on arguments and exceptions for a function are not available in the plugin. This request adds the annotation field to the Arguments struct to expose the annotations for plugins to use.

### CODE REVIEW

File: idl/internal/lex.rl [hunk 1]
```diff
@@ -4,9 +4,7 @@ package internal
 
 import (
     "fmt"
-    "io"
     "strconv"
-    "strings"
```
Comment (abhinav) [inline idl/internal/lex.rl]: 🤦‍♂ 
thanks for fixing

File: idl/internal/y.go [hunk 2]
```diff
@@ -1,5 +1,37 @@
 // @generated Code generated by yacc
 
+
+// @generated Code generated by yacc
+
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (abhinav) [inline idl/internal/y.go]: there's something wrong here. would you mind reverting the changes to this file?

File: plugin/api.thrift [hunk 3]
```diff
@@ -130,6 +130,24 @@ struct Argument {
...
+    /**
+     * Annotations defined on this argument.
+     *
+     * Given,
+     *
+     *   void setValue(
+     *   	1: SetValueRequest req
+     *   ) throws (
+     *		1: BadRequestError badRequestError (cache = "false")
+     *   )
```
Comment (abhinav) [inline plugin/api.thrift]: indentation is a bit off here with tabs/spaces. for the thrift file, we should use spaces only.

Comment (dmtsui) [reply plugin/api.thrift]: using spaces only now

File: idl/internal/y.go (hunk 2 above)
Comment (dmtsui) [reply idl/internal/y.go]: I've reset this as well as the lex.go file to the original state

File: plugin/api.thrift [hunk 4]
```diff
@@ -130,6 +130,24 @@ struct Argument {
...
+     *
+     * Given,
+     *
+     *   void setValue(
+     *     1: SetValueRequest req
+     *   ) throws (
+     *     1: BadRequestError badRequestError (cache = "false")
+     *   )
+     *
+     * The annotations will be,
```
Comment (abhinav) [inline plugin/api.thrift]:
```suggestion
     * The annotations for the Argument representing badRequestError will be,
```

**Comment by abhinav** [comment]
Oops!

**Pull request closed without merging**